applications/StructuralMechanicsApplication/tests/test_truss_adjoint_element_3d2n.py

```diff
@@ -0,0 +1,288 @@
+from __future__ import print_function, absolute_import, division
+import KratosMultiphysics
+
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import math
+
+def get_displacement_vector(mp,disp):
+    index=0
+    for node in mp.Nodes:
+        disp[index] = node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X,0)
+        index = index + 1
+        disp[index] = node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y,0)
+        index = index + 1
+        disp[index] = node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z,0)
+        index = index + 1
+ 
+def add_variables(mp):
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.DISPLACEMENT)
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.REACTION)
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.VOLUME_ACCELERATION)
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.VELOCITY)
+    mp.AddNodalSolutionStepVariable(KratosMultiphysics.ACCELERATION)
+
+def apply_material_properties(mp,dim):
+    #define properties
+    mp.GetProperties()[1].SetValue(KratosMultiphysics.YOUNG_MODULUS,1000)
+    mp.GetProperties()[1].SetValue(KratosMultiphysics.DENSITY,7850)
+    mp.GetProperties()[1].SetValue(StructuralMechanicsApplication.CROSS_AREA,10)
+    mp.GetProperties()[1].SetValue(StructuralMechanicsApplication.TRUSS_PRESTRESS_PK2,5)
+    g = [0,0,0]
+    mp.GetProperties()[1].SetValue(KratosMultiphysics.VOLUME_ACCELERATION,g)
+    cl = StructuralMechanicsApplication.TrussConstitutiveLaw()
+    mp.GetProperties()[1].SetValue(KratosMultiphysics.CONSTITUTIVE_LAW,cl)
+
+def copy_solution_step_data_of_node(node, mp_old, node_id, step=0):
+    node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X,step,
+    mp_old.Nodes[node_id].GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X,0))
+    node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y,
+    mp_old.Nodes[node_id].GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y,0))
+    node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z,
+    mp_old.Nodes[node_id].GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Z,0))
+
+def shape_perturbation_correction_factor(node_1, node_2):
+    dx = node_1.X - node_2.X
+    dy = node_1.Y - node_2.Y
+    dz = node_1.Z - node_2.Z
+    l = math.sqrt(dx*dx + dy*dy + dz*dz)
+    return l    
+  
+def zero_vector(size):
+    v = KratosMultiphysics.Vector(size)
+    for i in range(size):
+        v[i] = 0.0
+    return v
+
+def create_property_perturbed_elements(mp, delta, new_element_name):
```

Comment:
model_part

Reply:
whats the differnce between mp and the local model_part below? Try to avoid abbreviations for variables.

Reply:
this function establishes based on a passed model part (mp) a new (shape- or property-disturbed) model part (model_part). I renamed it in model_part and perturbed_model_part for better understanding.